src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
more efficient to use .some instead of .find method.  when you need to check if at least one element in an array meets a condition, .some is generally more performant than .find because .some stops iterating once it finds the first element that satisfies the condition